File: runtime/src/de/fatox/meta/entity/MetaEntityManager.java
package de.fatox.meta.entity;

import com.badlogic.gdx.utils.Array;
import de.fatox.meta.api.entity.EntityManager;

public class MetaEntityManager implements EntityManager<Meta3DEntity> {
    private Array<Meta3DEntity> entities = new Array<>();
    private Array<Meta3DEntity> staticEntities = new Array<>();

    public MetaEntityManager() {

    }

    @Override
    public Iterable<Meta3DEntity> getEntities() {
        return entities;
    }

    @Override
    public Iterable<Meta3DEntity> getStaticEntities() {
        return staticEntities;
    }

    @Override
    public void addEntity(Meta3DEntity entity) {
        entities.add(entity);
    }

    @Override
    public void addStaticEntity(Meta3DEntity entity) {
        staticEntities.add(entity);
    }

    @Override
    public void removeEntity(Meta3DEntity entity) {

    }

    @Override
    public void clear() {
        entities.clear();
        staticEntities.clear();
    }
}

File: runtime/src/main/kotlin/de/fatox/meta/entity/MetaEntityManager.kt
package de.fatox.meta.entity

import com.badlogic.gdx.utils.Array
import de.fatox.meta.api.entity.EntityManager

class MetaEntityManager : EntityManager<Meta3DEntity> {
	override val entities: Array<Meta3DEntity> = Array<Meta3DEntity>()
	override val staticEntities: Array<Meta3DEntity> = Array<Meta3DEntity>()

	override fun addEntity(entity: Meta3DEntity) {
		entities.add(entity)
	}

	override fun addStaticEntity(entity: Meta3DEntity) {
		staticEntities.add(entity)
	}

	override fun removeEntity(entity: Meta3DEntity) {
		staticEntities.removeValue(entity, true)
		entities.removeValue(entity, true)
	}

	override fun clear() {
		entities.clear()
		staticEntities.clear()
	}
}
